src/amp-story-player/amp-story-player-impl.js

```diff
@@ -730,20 +771,79 @@ export class AmpStoryPlayer {
 
   /**
    * Indicates the player changed story.
+   * @param {!Object} data
    * @private
    */
-  signalNavigation_() {
+  signalNavigation_(data) {
+    const event = createCustomEvent(this.win_, 'navigation', data);
+    this.element_.dispatchEvent(event);
+  }
+
+  /**
+   * Triggers when swithing from one story to another.
+   * @private
+   */
+  onNavigation_() {
     const index = this.currentIdx_;
     const remaining = this.stories_.length - this.currentIdx_ - 1;
-    const event = createCustomEvent(
-      this.win_,
-      'navigation',
-      dict({
-        'index': index,
-        'remaining': remaining,
-      })
-    );
-    this.element_.dispatchEvent(event);
+    const navigation = {
+      'index': index,
+      'remaining': remaining,
+    };
+
+    this.signalNavigation_(navigation);
+    this.maybeFetchMoreStories_(navigation);
+  }
+
+  /**
+   * Fetches more stories if appropiate.
+   * @param {!Object} data
+   */
+  maybeFetchMoreStories_(data) {
+    if (
+      data.remaining <= FETCH_STORIES_THRESHOLD &&
+      this.isFetchingEnabled_()
+    ) {
+      this.fetchStories_()
+        .then((stories) => {
+          this.add(stories);
+        })
+        .catch((reason) => {
+          console /*OK*/
+            .error(`[${TAG}] `, reason);
+        });
+    }
+  }
+
+  /**
+   * Checks if fetching more stories is enabled and validates the configuration.
+   * @return {boolean}
+   * @private
+   */
+  isFetchingEnabled_() {
+    const {behavior} = this.readPlayerConfig_();
+
+    if (!behavior) {
+      return false;
+    }
+
+    const isBehaviorDef = (behavior) => behavior.on && behavior.action;
+
+    const hasEndFetchBehavior = (behavior) =>
+      behavior.on === 'end' &&
+      behavior.action === 'fetch' &&
+      behavior.endpoint.length > 0;
+
+    if (
+      isBehaviorDef(behavior) &&
+      hasEndFetchBehavior(behavior) &&
+      this.shouldFetch_
+    ) {
+      this.shouldFetch_ = false;
+      return true;
+    }
+
+    return false;
```

Comment:
If I read this correctly, it only fetches once and then never again? We should keep polling for more stories all the time.

This whole block should not be done on each navigation. It could be done at build time or on first navigation. If on navigation, it should be done only once and cached.